Documentation updates

nickerso opened the pull request:
This is a work in progress to address a range of documentation issues: 

- [ ] #315 Clarify parts of the coding guidelines
- [ ] #258 Add information about SWIG and Python
- [ ] #221 Improve development setup documentation
- [ ] #149 Figure numbers not showing in documentation
- [ ] #373 Documentation for clang-tidy

- [ ] #425 Tutorials for libCellML end users

kerimoyle committed:
Update dev_setup.rst

Making the link to Windows installer for libxml2 more obvious.

kerimoyle committed:
Update dev_building.rst

kerimoyle committed:
WIP updating windows instructions

kerimoyle committed:
WIP updating dev instructions for windows

kerimoyle committed:
WIP

kerimoyle committed:
Create libCellMLBuilding-CMakeOptionsInRed.png

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
WIP - including tables of options

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Merge branch 'documentation' of https://github.com/kerimoyle/libcellml-keri into documentation

kerimoyle committed:
Delete dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Merge branch 'documentation' of https://github.com/kerimoyle/libcellml-keri into documentation

kerimoyle committed:
Create dev_building2.rst

kerimoyle committed:
WIP adding figures and testing

kerimoyle committed:
WIP

kerimoyle committed:
Update dev_building2.rst

kerimoyle committed:
v3

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Adding sphinx instructions

kerimoyle committed:
Update dev_setup.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Moving to main building page

nickerso committed:
Starting work on updating the developer documentation for the project.

This incomplete work is aimed at improving the documentation overall, mostly with a focus on helping potential contributors get this up and running. I have also added some placeholder text for when we have installers ready for users.

nickerso committed:
adding links to moved developer documents.

kerimoyle committed:
Update dev_setup.rst

kerimoyle committed:
Added python links to requirements for setup

kerimoyle commented:
Changes to dev_setup and dev_building pages as in #306 are here: 
https://github.com/kerimoyle/libcellml-keri/tree/documentation/docs

kerimoyle committed:
starting to write glossary terms and links

kerimoyle committed:
Getting refs to work

kerimoyle committed:
One sentence per line

kerimoyle committed:
Glossary tweaks

I'm not sure how best to format the glossary page: If there is more than one line (because of the one sentence per line thing) then the first sentence is rendered in bold and looks horrible ...

kerimoyle committed:
WIP playing - only committing to switch

The css tag for the first line underneath a definition is set as <dt> which makes it the start of a new list, and therefore title case.  It doesn't seem to happen in the readthedocs version so might be something different between my build and theirs which affects formatting.

nickerso committed:
Merge remote-tracking branch 'keri/documentation' into doc-updates

nickerso committed:
Merge branch 'develop' into doc-updates

nickerso commented:
I have merged the changes from #335 into this one, so lets make this the main PR for updating the docs.

nickerso committed:
Clean and tidy following restructure.

Updating image paths for new locations, correcting some internal cross references, trying to remove all the sphinx-build warnings.

kerimoyle committed:
Merge branch 'develop' into doc-updates

hsorby committed:
Merge in changes from 'prime/develop' fixing conflicts.
